### PR TITLE
Add support for HTTPS

### LINKED ISSUE #202: Does kernel gateway support HTTPS connections?
The documentation [1] does not list options that correspond with `NotebookApp.keyfile`, `NotebookApp.certfile` or `NotebookApp.client_ca` [2] to support encrypted communication [3].

[1] https://jupyter-kernel-gateway.readthedocs.io/en/latest/config-options.html
[2] http://jupyter-notebook.readthedocs.io/en/latest/config.html
[3] http://jupyter-notebook.readthedocs.io/en/latest/public_server.html#using-ssl-for-encrypted-communication

### PR DESCRIPTION
**Pull request by parente**
Fix #202. Cleanup some nits in gatewayapp.py along the way.

### CODE REVIEW

File: kernel_gateway/gatewayapp.py [hunk 1]
```diff
@@ -395,17 +440,44 @@ def init_webapp(self):
             if kg_name not in self.web_app.settings:
                 self.web_app.settings[kg_name] = trait_value.get(obj=self.personality)
             else:
-                self.log.warning('The personality trait name, %s, conflicts with a kernel gateway trait.' , trait_name)
+                self.log.warning('The personality trait name, %s, conflicts with a kernel gateway trait.', trait_name)
+
+    def _build_ssl_options(self):
+        """Build a dictionary of SSL ptions for the tornado HTTP server.
```
Comment (willingc) [inline kernel_gateway/gatewayapp.py]: options

Comment (blink1073) [inline kernel_gateway/gatewayapp.py]: Sounds like a new CERN discovery.

**Comment by willingc** [comment]
@parente If you will rebase, I'll merge it in. Thanks @rgbkrk.

**Comment by parente** [comment]
Rebased and removed all traces of the mythical p-tion.

**Comment by parente** [comment]
Tests passing after @minrk 's forced update of pip in #218 for Travis. GTG.

**Comment by minrk** [comment]
Huzzah!

**Comment by parente** [comment]
Thanks again for the reviews, everyone.

**Pull request merged**